compiler/tests/generated/alias_with_allocator_types.h

```diff
@@ -46,6 +46,22 @@ namespace test
         foo(const foo&) = default;
 #endif
         
+        foo(const foo& other, const arena& allocator)
```

Comment:
Should these constructors be templated so that they are only expanded if they are actually used?